beacon_node/beacon_chain/src/block_verification.rs

```diff
@@ -810,7 +814,7 @@ impl<T: BeaconChainTypes> GossipVerifiedBlock<T> {
             });
         }
 
-        let block_root = get_block_root(&block);
+        let block_root = get_block_header_root(block_header);
```

Comment:
I was wondering if we could delete `get_block_root` but it looks like it's used in `RpcBlock`, so happy to leave it as-is for now.

Aside: I'm not super familiar with `RpcBlock` but have we checked that blocks that fail availability checks are passed to the slasher?

Reply:
blocks are added to the slasher on calling `BeaconChain::process_block` which calls `IntoExecutionPendingBlock::into_execution_pending_block` here
https://github.com/sigp/lighthouse/blob/a3fb27c99b6696b7b9ccf69d5a43977f44440687/beacon_node/beacon_chain/src/block_verification.rs#L739-L756

`RpcBlock` also implements `IntoExecutionPendingBlock` so it's added to the slasher regardless if it passes or fails the availability checks